src/Mixins/Expectation.php

```diff
@@ -714,14 +714,18 @@ public function toBeWritableFile(): Expectation
      *
      * @return Expectation<TValue>
      */
-    public function toMatchArray(iterable|object $array): Expectation
+    public function toMatchArray(iterable $array): Expectation
     {
         if (is_object($this->value) && method_exists($this->value, 'toArray')) {
             $valueAsArray = $this->value->toArray();
         } else {
             $valueAsArray = (array) $this->value;
         }
 
+        if (is_object($array) && method_exists($array, 'toArray')) {
```

Comment:
This need code needs a test.

Reply:
I removed that code, added it under the wrong assumption that the expectation should accept an object